x/posts/internal/keeper/keeper.go

```diff
@@ -139,7 +152,23 @@ func (k Keeper) GetPostsFiltered(ctx sdk.Context, params types.QueryPostsParams)
 			matchCreator = params.Creator.Equals(post.Creator)
 		}
 
-		if matchParentID && matchCreationTime && matchAllowsComments && matchSubspace && matchCreator {
+		// match hashtags if provided
+		if len(params.Hashtags) > 0 {
+			postHashtags := k.GetPostHashtags(post)
+			if len(postHashtags) == len(params.Hashtags) {
+				sort.Strings(postHashtags)
+				sort.Strings(params.Hashtags)
+				for index, hashtag := range params.Hashtags {
+					if matchHashtags = postHashtags[index] == hashtag; !matchHashtags {
```

Comment:
There's something sketchy about this equal check and the `GetPostHashtags` implementation. 

Suppose we have: 

- Message: `This is a #test post` 
- Query: `posts?hashtags=test`

Now, `GetPostHashtags` returns the following array: `[#test]` so, `#test` is matched against `test`, which is not the same and thus false is returned. 

This means that the post is not included inside the list, which is incorrect. 

What I think should happen is that `GetPostHashtags` should return the list of hashtags **without** the leading `#`. This would make for easier comparison as well.